{
  "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/25753",
  "repository_url": "https://api.github.com/repos/ethereum/go-ethereum",
  "labels_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/25753/labels{/name}",
  "comments_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/25753/comments",
  "events_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/25753/events",
  "html_url": "https://github.com/ethereum/go-ethereum/issues/25753",
  "id": 1371468418,
  "node_id": "I_kwDOAOvK985RvvKC",
  "number": 25753,
  "title": "Geth 1.10.23 fresh sync is broken for Post-Merge networks with error: Beacon client online, but never received consensus updates.",
  "user": {
    "login": "begetan",
    "id": 17225934,
    "node_id": "MDQ6VXNlcjE3MjI1OTM0",
    "avatar_url": "https://avatars.githubusercontent.com/u/17225934?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/begetan",
    "html_url": "https://github.com/begetan",
    "followers_url": "https://api.github.com/users/begetan/followers",
    "following_url": "https://api.github.com/users/begetan/following{/other_user}",
    "gists_url": "https://api.github.com/users/begetan/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/begetan/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/begetan/subscriptions",
    "organizations_url": "https://api.github.com/users/begetan/orgs",
    "repos_url": "https://api.github.com/users/begetan/repos",
    "events_url": "https://api.github.com/users/begetan/events{/privacy}",
    "received_events_url": "https://api.github.com/users/begetan/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [
    {
      "id": 72233650,
      "node_id": "MDU6TGFiZWw3MjIzMzY1MA==",
      "url": "https://api.github.com/repos/ethereum/go-ethereum/labels/type:bug",
      "name": "type:bug",
      "color": "FF5E5E",
      "default": false,
      "description": ""
    }
  ],
  "state": "open",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 0,
  "created_at": "2022-09-13T13:33:39Z",
  "updated_at": "2022-09-13T13:33:39Z",
  "closed_at": null,
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "#### System information\r\n\r\nGeth version:\r\n```\r\nVersion: 1.10.23-stable\r\nGit Commit: d901d85377c2c2f05f09f423c7d739c0feecd90a\r\nGit Commit Date: 20220824\r\nArchitecture: arm64\r\nGo Version: go1.18.5\r\nOperating System: linux\r\n```\r\nPrysm beacon-chain version:\r\n```\r\nbeacon-chain version Prysm/v3.1.1/653ea3b030c3bb99aa3f3d95a8de9f4d6e147930. Built at: 2022-09-09 17:24:18+00:00\r\n```\r\n\r\n#### Expected behaviour\r\nWe expect system should start syncing from scratch both for Geth and Prysm beacon clients\r\n\r\n\r\n#### Actual behaviour\r\nPrysm is syncing: [log1](https://gist.github.com/begetan/ea935fe9c26a4a269d7458b0cfec37eb)\r\nGeth is not able to sync: [log2](https://gist.github.com/begetan/10733dbd611fb85da61030677d2f315e)\r\n\r\nGeth not able to sync with this error message:\r\n`Beacon client online, but never received consensus updates. Please ensure your beacon client is operational to follow the chain!` \r\n\r\n#### Steps to reproduce the behaviour\r\n\r\n1. Configure geth with systemd unit\r\n2. Configure prysm with systemd unit\r\n3. Create jwt token\r\n3. Start services:\r\n```\r\nsystemctl enable geth.service --now\r\nsystemctl enable prysm.service --now\r\n\r\n```\r\nThis issue is not fully repeatable. It may depend on the startup timings.\r\nIt occurs at  90% tries of deployment of Goerli and 50% of Ropsten deploynment. Mainnet is not affected at the moment, because this check is in Post-Merge code.\r\n\r\nAnother important thing, get in this status provides incorrect sync state:\r\n\r\n```\r\ncurl -s -X POST -H \"Content-Type:application/json\" --data '{\"jsonrpc\":\"2.0\",\"method\":\"eth_syncing\",\"id\":1}' localhost:8545\r\n{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":false}\r\n\r\ndu -shc /data/*\r\n2.2M    /data/geth\r\n4.0K    /data/jwtsecret\r\n325M    /data/prysm\r\n327M    total\r\n```\r\nYou can see that geth data directory is empty at the moment of syncing=false report.  We didn't success to recover geth from dead-loop by restarting geth of prysm services. The only way to recover this situation is to replace geth with binary version 1.10.21\r\n",
  "closed_by": null,
  "reactions": {
    "url": "https://api.github.com/repos/ethereum/go-ethereum/issues/25753/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/ethereum/go-ethereum/issues/25753/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
[

]
